packages/grid/_modules_/grid/models/gridExport.ts

```diff
@@ -23,6 +23,16 @@ export interface GridExportCsvOptions {
    * @default false
    */
   utf8WithBom?: boolean;
+  /**
+   * The columns exported in the CSV
+   * This should only be used if you want to restrict the columns exports
+   */
+  columnKeys?: string[];
```

Comment:
I think it will be better to just have a value like `notExportable` or `disableExport` inside the `ColDef` and use it to know what column is exportable and what not.

Reply:
https://github.com/mui-org/material-ui-x/issues/1435#issuecomment-825492594

For me that was the discussion here.
If we follow the AG Grid solution, the "not exportable" columns are flagged with a param in the export method (which will also be available as in the prop `csvOptions` in `GridToolbarExport`)
If we follow the material-table solution, the "not exportable" columns are flagged with a param in the `ColDef`

I tend to prefer the AG Grid approach here because it easily allow to to several export buttons with different configurations. Which is a use case that I had quite often on my previous job.

---

Of course we could also implement both (right now or later).

Reply:
That comment is regarding the delimiter as far as I can see. That to me is a different use case. 

I'm not saying that we should remove what you have done here but rather add the option in the `ColDef` as well - logic is that there are certain columns that we can be 100% sure that devs wouldn't want to export such as the checkbox or ID or if they have a column with buttons that do different actions.
Since the CSV export is a free feature it makes sense to provide the option to mark those columns as not exportable somehow in the definition.

Reply:
I mis-read the comment then.

> Since the CSV export is a free feature it makes sense to provide the option to mark those columns as not exportable somehow in the definition.

Do we agree that the "AG Grid" solution (= the current state of the PR) does allow the free user to use it (by passing a prop to `GridToolbarExport` ? Or am I missing something ?

---

And no problem to implement both.
I suppose that if both were used at the same time, the `columnKeys` would have the priority over the `ColDef`

Reply:
Done :+1: